src/cr/cube/crunch_cube.py

```diff
@@ -795,13 +795,13 @@ def scale_means(self, hs_dims=None, prune=False):
     @lazyproperty
     def univariate_ca_main_axis(self):
         """For univariate CA, the main axis is the categorical axis"""
-        dim_types = [dim.type for dim in self.dimensions]
+        dim_types = [d.dimension_type for d in self.dimensions]
         return dim_types.index('categorical')
 
     def valid_indices_with_selections(self, include_missing=False):
         """Get all valid indices (including MR selections)."""
         return [
-            dim.valid_indices(include_missing)
+            dim.element_indices(include_missing)
```

Comment:
I renamed `Dimension.valid_indices()` to `.element_indices()` since it can be used to get both all and only valid indices. This eliminates a jarring contradiction in the naming.